packages/core/src/modules/interest-group/interest-group-repository.ts

```diff
@@ -28,9 +30,21 @@ export class InterestGroupRepositoryImpl implements InterestGroupRepository {
     return interestGroup ? mapToInterestGroup(interestGroup) : undefined
   }
 
-  async getAll(pageable: Pageable): Promise<Collection<InterestGroup>> {
-    const res = paginatedQuery(this.db.selectFrom("interestGroup").selectAll(), pageable, mapToInterestGroup)
-    return res
+  async getAll(pageable: Pageable) {
+    const query = this.db.selectFrom("interestGroup").selectAll()
+
+    const result = await singleColPaginatedQuery(query, {
+      pageable,
+      decodeCursor: decodeUlidIdCursor,
+      buildCursor: buildUlidIdCursor,
```

Comment:
We can just derive these from specifying which columns to paginate on, no?

Reply:
Maybe, but I think that adds significant type complexity

Reply:
I think losing typing for a better API is a worthwhile tradeoff

Reply:
If we drop ulids and have id+created_at for everything then we can default it to that ,and have an option for overriding

Reply:
sounds good

Reply:
I implemented sorting by id by default now. Thoughts?

Reply:
I just realized that this implementation just recurses infinitely. On it.

Reply:
There, now it works. Had to do some funky not null assertions though